runtime/stdlib/contracts/crypto.cdc

```diff
@@ -1,16 +1,4 @@
 
-pub struct interface SignatureVerifier  {
-
-    pub fun verify(
-        signature: [UInt8],
-        tag: String,
-        signedData: [UInt8],
-        publicKey: [UInt8],
-        signatureAlgorithm: SignatureAlgorithm,
-        hashAlgorithm: HashAlgorithm
-    ): Bool
-}
-
 pub struct interface Hasher  {
```

Comment:
Out of curiosity, does the `hash` function need to be defined in an interface ? Isn't easier to define a public function `hash` within the contract, and avoid the private hasher field in the contract?

Reply:
No, it could just be a function. I declared it in an interface, because the interface is more flexible/extensible, e.g. we could have easily added more functions.